schemas/ispyb/stored_programs/sp_retrieve_screenings_for_sample.sql

```diff
@@ -0,0 +1,26 @@
+DELIMITER ;;
+CREATE OR REPLACE DEFINER= `ispyb_root`@`%` PROCEDURE `retrieve_screenings_for_sample`(p_id int unsigned)
+COMMENT 'Returns multi-row result set with the screening for the given sampleId'
+BEGIN
+	IF p_id IS NOT NULL THEN
+		SELECT DISTINCT
+		  scr.screeningId "screeningId",
+		  scr.dataCollectionId "dataCollectionId",
+		  scr.bltimeStamp "bltimeStamp",
+		  scr.programVersion "programVersion",
+		  scr.comments "comments",
+		  scr.shortComments "shortComments",
+		  scr.diffractionPlanId "diffractionPlanId",
+		  scr.dataCollectionGroupId "dataCollectionGroupId",
+		  scr.xmlSampleInformation "xmlSampleInformation"
+		FROM BLSample bls
+		  INNER JOIN BLSubSample bss ON bls.blSampleId = bss.blSampleID
+		  INNER JOIN DataCollection dc ON bss.blSubSampleId = dc.blSubSampleID
+		  INNER JOIN Screening scr ON dc.dataCollectionId = scr.dataCollectionId
```

Comment:
I could be wrong, but I didn't think normal MX beamlines were using the `BLSubSample` table? If that is a mistake, then the joins can be rewritten as:

		FROM BLSample bls
		  INNER JOIN DataCollection dc ON bls.blSampleId = dc.blSampleId
		  INNER JOIN Screening scr ON dc.dataCollectionId = scr.dataCollectionId

Reply:
I think you're probably right.